xbmc/Application.cpp

```diff
@@ -1809,6 +1809,8 @@ bool CApplication::LoadUserWindows()
               id = atol(pType->FirstChild()->Value());
           }
           std::string visibleCondition;
+          for (TiXmlElement* node = pRootElement->FirstChildElement("visible"); node; node = node->NextSiblingElement("visible"))
+            g_SkinInfo->ResolveIncludes(pRootElement->FirstChildElement("visible"));
```

Comment:
>I expected CGUIWindow::Load(pRootElement) to handle the window visibility for custom windows like it does regular windows

I thought the same and we definitely need to change something here. I've already started to work on it. After that every other comment should therefore become obsolete.